### DISCUSSION TITLE
Password reset email not sending to valid email addresss.

### DISCUSSION THREAD

**Discussion by Benjamin-Hayduchok** [pocketbase/pocketbase#2724] [Q&A]
I am testing the password reset functionality and have attempted a password reset on multiple email address I have access to. I read in one of the other discussions that pocketbase uses the sendmail unix command for sending email and it may not be reliable.

Is that the case, and is there anything I can do about it? To be clear, I have attempted the password reset API endpoint multiple times, and I get a valid 204 back. I see the request in the logs that it went through, but I have never gotten even one email yet on either email address.

I would like users to be able to reset password if they need to. Are there any work arounds for this, or should I try an OAuth2 approach for more reliability?

If there is any additional information I can provide, I would be more than happy. Thank you so much in advance!

**Comment by ganigeorgiev** [comment]
It is recommended to use a SMTP server if you want to send emails reliably. Note that even with properly configured `sendmail` setup most mail servers will directly reject the sent email (_you can inspect your sendmail logs with something like `cat /var/spool/mail/{user}`_).

For test purposes I often use [mailtrap.io](https://mailtrap.io/).

**Reply by danilomsilva** [reply]
Hi there, I get this error message message even using an SMTP server such as Brevo and getting a 200 after making a request to /request-password-reset.

```
{ code: 400,
  message: 'An error occurred while loading the submitted data.',
  data: {}
  }
```

I tried sending an test email using pocketbase UI and I got the test email, however it is not sending from my code even though I think it is correct.

```// /api/collections/usuario/request-password-reset
export async function requestPasswordChange(email: string) {
  const response = await fetch(
    `${process.env.BASE_API_URL}/collections/usuario/request-password-reset`,
    {
      method: 'POST',
      headers: {
        'Content-type': 'application/json',
      },
      body: JSON.stringify(email),
    }
  );

  const data = await response.json();
  return data;
}```

Could you please help me? I might be missing something on pocketbase side?
Thanks a million!

**Reply by ganigeorgiev** [reply]
@danilomsilva At first glance it seems that you are not sending a proper body - try replacing `JSON.stringify(email)` with `JSON.stringify({ email })`.
But in any case your issue is not related to the above discussion. If you are not able to resolve it, please open a new one with more detail about your environment and some steps to reproduce.
You can also start the executable with `--dev` and/or check the logs in the _Admin UI > Logs_ for a more detailed error message.

**Reply by danilomsilva** [reply]
Hi, I can see what's going on now after reading the logs. 
```"errorMessage": "You've already requested a password reset."```

Thanks a lot for the help!!
